learning_journal/learning_journal/tests.py

```diff
@@ -0,0 +1,161 @@
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture(scope="function")
+def populated_db(request, sqlengine):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(MyEntry(title="Test Title", body="This is a test entry, James is being awesome.", creation_date=datetime.datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(MyEntry).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Call a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_home_view(dummy_request, new_session):
+    """Test entries are in home view."""
+    from .views.default import home_view
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    info = home_view(dummy_request)
+    assert "entries" in info
+
+
+def test_detail_view(new_session):
+    """Test detail view has a body."""
+    from .views.default import detail_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
+    info = detail_view(request)
+    assert "this is a test" in info['entry'].body
+
+
+def test_create_view():
+    """Test create view."""
+    from .views.default import create_view
+    request = testing.DummyRequest()
+    create_view(request)
+    assert request.response.status_code == 200
+
+
+def test_edit_view(new_session):
+    """Test update view has a body."""
+    from .views.default import edit_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
+    info = edit_view(request)
+    assert info["entry"].body == "this is a test"
+
+# -------Functional Tests----------
+
+
+@pytest.fixture()
+def testapp(sqlengine, setup_test_env):
+    """Setup TestApp."""
+    from learning_journal import main
+    app = main({}, **DB_SETTINGS)
+    from webtest import TestApp
+    return TestApp(app)
+
+
+def test_layout_root_home(testapp, populated_db):
+    """Test layout root of home route."""
+    response = testapp.get('/', status=200)
+    assert b'Test Title' in response.body
+
+
+def test_layout_root_create(testapp):
+    """Test layout root of create route."""
+    response = testapp.get('/create', status=200)
+    assert response.html.find("textarea")
+
+
+def test_layout_root_edit(testapp, populated_db):
+    """Test layout root of edit route."""
+    response = testapp.get('/edit/1', status=200)
+    html = response.html
+    assert html.find("h2")
```

Comment:
I'd love to see you create more extensive and comprehensive tests for the layout you've got.  Or at least to ensure that you are testing the most important parts.  Is there a form there?  Does it have the right inputs?  The right names on those inputs?  If you're viewing the edit page, are the right values present when the page renders?